tests/op_datasets/etl/intermediate/models/test_daily_address_summary.py

```diff
@@ -1,108 +1,322 @@
-    assert pl.select(df_multi_txs["l2_contrib_gas_fees"]).item() == Decimal(
-        str(0.0000035911200000000)
-    )
-    assert pl.select(df_multi_txs["l1_contrib_gas_fees"]).item() == Decimal(
-        str(0.0000001497346364060)
-    )
+INPUT_TEST_DATA = InputTestData.at(__file__)
+
+
+class TestDailyAddressSummary001(IntermediateModelTestBase):
+    model = "daily_address_summary"
+    inputdata = InputTestData.at(__file__)
+    chains = ["op"]
+    dateval = date(2024, 10, 1)
+    datasets = ["blocks", "transactions"]
+    block_filters = [
+        "{block_number} = 126078000",
+        "{block_number} = 126073799",
+        "{block_number} = 126090026",
+        "{block_number} % 1000 <= 2",
+    ]
+
+    _enable_fetching = True
+
+    def test_system_address_not_included(self):
+        ans = self._duckdb_client.sql(f"""
+        SELECT * FROM daily_address_summary WHERE address = '{SYSTEM_ADDRESS}'
+        """)
+
+        assert len(ans) == 0
+
+    def test_uniqueness(self):
+        unique_count = self._duckdb_client.sql("""
+        SELECT COUNT(*) FROM (
+            SELECT DISTINCT address, chain_id, dt FROM daily_address_summary
+        )
+        """)
+        total_count = self._duckdb_client.sql("""
+        SELECT COUNT(*) FROM daily_address_summary
+        """)
+
+        unique_count_val = unique_count.fetchone()[0]
+        total_count_val = total_count.fetchone()[0]
+
+        assert unique_count_val == 785
+        assert unique_count_val == total_count_val
+
+    def test_model_schema(self):
+        schema = (
+            self._duckdb_client.sql("DESCRIBE daily_address_summary")
+            .pl()
+            .select("column_name", "column_type")
+            .to_dicts()
+        )
+        actual_schema = {row["column_name"]: row["column_type"] for row in schema}
+
+        assert actual_schema == {
+            "dt": "DATE",
+            "chain": "VARCHAR",
+            "chain_id": "INTEGER",
+            "address": "VARCHAR",
+            "total_txs": "BIGINT",
+            "total_txs_success": "BIGINT",
+            "total_blocks": "BIGINT",
+            "total_blocks_success": "BIGINT",
+            "min_block_number": "BIGINT",
+            "max_block_number": "BIGINT",
+            "block_interval_active": "BIGINT",
+            "min_nonce": "BIGINT",
+            "max_nonce": "BIGINT",
+            "nonce_interval_active": "BIGINT",
+            "min_block_timestamp": "UINTEGER",
+            "max_block_timestamp": "UINTEGER",
+            "time_interval_active": "UINTEGER",
+            "unique_hours_active": "BIGINT",
+            "num_to_addresses": "BIGINT",
+            "num_to_addresses_success": "BIGINT",
+            "num_method_ids": "BIGINT",
+            "total_l2_gas_used": "DECIMAL(38,0)",
+            "total_l2_gas_used_success": "DECIMAL(38,0)",
+            # TODO: Should l1_gas_used values be DECIMAL(38,0) instead of DOUBLE ???
```

Comment:
@chuxinh  I'll need your input on this question. Using the current model definition the types for `total_l1_gas_used` and `total_l1_gas_used_success` are coming out as `DOUBLE`.  I wonder if we need to update the definition to make sure it comes out as decimal or bigint.

Reply:
oh maybe because it's calculated that's why. I think we should keep it consistent as `DECIMAL(38, 0)`

Reply:
cool, then maybe there is something we have to adjust in the expression so that it never coerces to a DOUBLE. My guess is that the division operations are causing things to become double.  Instead of dividing one can instead multiply by the equivalent DECIMAL value.